crates/swc_ecma_parser/src/parser/stmt/module_item.rs

```diff
@@ -81,30 +72,56 @@ impl<I: Tokens> Parser<I> {
                 specifiers: vec![],
                 type_only: false,
                 with,
+                phase: Default::default(),
             })));
         }
 
+        // https://github.com/microsoft/TypeScript/issues/56376
         let type_only = self.input.syntax().typescript()
-            && is!(self, "type")
-            && (peeked_is!(self, '{') || !peeked_is!(self, "from") && !peeked_is!(self, ','));
-
-        if type_only {
-            assert_and_bump!(self, "type");
+            && (peeked_is!(self, '{')
+                || !peeked_is!(self, "from") && !peeked_is!(self, ',') && !peeked_is!(self, '='))
+            && eat!(self, "type");
 
-            if is!(self, IdentRef) && peeked_is!(self, '=') {
-                return self
-                    .parse_ts_import_equals_decl(
-                        start, /* is_export */ false, /* is_type_only */ true,
-                    )
-                    .map(ModuleDecl::from)
-                    .map(ModuleItem::from);
-            }
+        if self.input.syntax().typescript() && is!(self, IdentRef) && peeked_is!(self, '=') {
+            return self
+                .parse_ts_import_equals_decl(
+                    start, /* is_export */ false, /* is_type_only */ type_only,
+                )
+                .map(ModuleDecl::from)
+                .map(ModuleItem::from);
         }
 
         let mut specifiers = vec![];
 
+        let mut phase = Default::default();
+        let mut default_ident = None;
+
         if is!(self, BindingIdent) {
-            let local = self.parse_imported_default_binding()?;
+            // maybe phase
+            let ident = self.parse_imported_default_binding()?;
+
+            // OK: import <phase> x from "foo";
+            // OK: import <phase> from from "foo";
+            // NO: import <phase> from "foo";
+            // NO: import <phase> from 'foo';
+            // OK: import <phase> { x } from "foo";
+            // OK: import <phase> * as T from "foo";
+            // NO: import <phase> "foo";
+            let is_phase = is_one_of!(self, BindingIdent, '{', '*')
+                && (!is!(self, "from") || peeked_is!(self, "from"));
```

Comment:
Is it possible to peek more than one token?
https://github.com/oxc-project/oxc/blob/891d353adeefbd54e56c2ff3e89a49697da0bc5b/crates/oxc_parser/src/js/module.rs#L457

Reply:
I'll add a method to do so. Also, we need to roll-up AST-breaking changes, so I'll work on #8026 and the setter prop AST thing.